docs/src/introduction_to_catalyst/advanced_intro_to_catalyst.md

```diff
@@ -0,0 +1,356 @@
+- When algebraic expression of species and/or parameters must be formed (as used e.g. [here](@ref ref)). Typically, these situations indicate that an [*observable*](@ref dsl_advanced_options_observables) should be created to designate the quantity of index.
+
+Generally, symbol-based designates is used when [models are created using the DSL](@ref advanced_intro_to_catalyst_modelling_approaches_DSL).
+
+### [Symbolics variables-based designation](@id advanced_intro_to_catalyst_designations_symbolics)
+Previously, we showed how model creation using ModelingToolkit involves [explicitly declaring variables and parameters, which then can be used to designate their values](@ref advanced_intro_to_catalyst_background_mtk). We also showed how Catalyst `ReactionSystem`s can be [created programmatically, in a similar way to how ModelingToolkit models are created](@ref advanced_intro_to_catalyst_modelling_approaches_programmatic). Here, just like for ModelingToolkit, we can represent species and parameters using their respective symbolic variables directly:
+```@example advanced_intro_to_catalyst_designations
+t = default_t()
+@species X(t) 
+@parameters p d
+rxs = [
+    Reaction(p, nothing, [X]), 
+    Reaction(d, [X], nothing)
+]
+rs_prog = ReactionSystem(rxs, t; name=:rs)
+rs_prog = complete(rs_prog)
+
+u0_symb = [X => 1.0]
+ps_symb = [p => 1.0, d => 0.2]
+oprob_symb = ODEProblem(rs_prog, u0_symb, tspan, ps_symb)
+sol_symb = solve(oprob_symb)
+nothing # hide
+```
+We can check that the resulting solution is the same as when we used symbol-based designation (and DSL-based model creation):
+```@example advanced_intro_to_catalyst_designations
+sol_syms == sol_symb
+```
+We can also use symbolic variables to e.g. index the output solution:
+```@example advanced_intro_to_catalyst_designations
+sol_symb[X]
+```
+
+The advantage with the Symbolics for is that we can form algebraic expressions from our symbolic variables. E.g. if we (for some reason) would like to evaluate the expression $2X + p$ across the solution vector, we can do this directly using the Symbolic form:
+```@example advanced_intro_to_catalyst_designations
+sol_symb[2X + p]
+```
+Indeed, we can confirm that this returns the expected output:
+```@example advanced_intro_to_catalyst_designations
+sol_symb[2X + p] == 2*sol[X] .+ 1.0
+```
+
+Symbolic variables-based representation is strictly superior to symbol-based representation. However, it is generally only used in combination with programmatic modelling (as during DSL-based modelling we never actually declare the symbolic variables, and thus cannot use them directly). To use symbolic variables-based representation in combination with the DSL, the [`@unpack` macro can be used](@ref ref).
+
+### [System-based symbolic designation](@id advanced_intro_to_catalyst_designations_sys_symb)
```

Comment:
The new section on how to access content of models got a more extensive description of this.